megamek/src/megamek/client/ui/swing/util/UIUtil.java

```diff
@@ -1187,4 +1187,63 @@ public static void updateWindowBounds(Window window) {
         window.setLocation(location);
         window.setSize(size);
     }
+
+    /**
+     * search for searchPattern in text starting from startPos in direction searchDown
+     * if search hits text start or end wrap until hit startPos
+     * return the position of the match if found
+     * return -1 if not found
+     */
+    public static int findPattern (String text, String searchPattern, int startPos, boolean searchDown) {
+        boolean found = false;
+        int len = text.length() - searchPattern.length();
+        int currentPos = 0;
+        if (searchDown) {
+            currentPos = startPos + 1;
+        }
+        else {
+            currentPos = startPos - searchPattern.length() - 1;
+
+            if (currentPos < 1) {
+                currentPos = len;
+            }
+        }
+
+        while (startPos != currentPos) {
+            String match = text.substring(currentPos, currentPos + searchPattern.length());

```

Comment:
Can't you just replace this whole thing with text.indexOf(searchPattern, startindex)? I'm all for rolling your own algorithms, but this is a solved problem, and I don't think we need a home-grown O(n*m) solution here.

Reply:
Searching backwards can be accomplished by using lastIndexOf(searchPattern, startIndex)